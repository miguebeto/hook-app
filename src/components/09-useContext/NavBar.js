import React from 'react';
import { NavLink } from 'react-router-dom';

//creamos los link que van a redirigir la pagina y harÃ¡n el match con los router
export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <div className="container-fluid">

            <NavLink className="navbar-brand" to="/home">useContext</NavLink>

            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
            <div className="navbar-nav">
                <NavLink exact activeClassName="active" to="/" className="nav-link" aria-current="page">Home</NavLink>
                <NavLink exact activeClassName="active" to="/about" className="nav-link">About</NavLink>
                <NavLink exact activeClassName="active" to="/login" className="nav-link">Login</NavLink>
            </div>
            </div>
        </div>
    </nav>
  )
};
