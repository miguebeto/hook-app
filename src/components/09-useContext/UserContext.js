import { createContext } from 'react';

//creamos el contexto que ser√° utilizadoo en el proveedor y en el consumidor
export const UserContext = createContext(null);