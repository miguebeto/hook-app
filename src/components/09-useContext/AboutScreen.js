import React, { useContext } from 'react';
import { UserContext } from './UserContext';


export const AboutScreen = () => {

  //extraemos el user y la funcion de seteo setUser que nos manda el proveedor en el main app
  const {user, setUser} = useContext(UserContext);

  //creamos la funcion que setea el estado a un objeto vacio cuando se da click en el botÃ³n
  const handleClick = () => {
    setUser({});
  }

  return (
       <div>
          <h1>About Screen</h1>
          <hr />
          <pre>
            { JSON.stringify(user, null, 3) }
          </pre>
          <button 
          className='btn btn-warning'
          onClick={ handleClick }
          >
             Logout
          </button>
      </div>
  )
};
