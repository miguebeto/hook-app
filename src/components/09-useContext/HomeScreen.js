import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => { 
  //extraemos el estado del context mediante el useContext y lo mostramos como objeto en pantalla con la etiqueta <pre>
  const {user} = useContext(UserContext);
  console.log(user);
  
  return (
      <div>
          <h1>Home Screen</h1>
          <hr />

          <pre>
            { JSON.stringify( user, null, 3 ) }
          </pre>
      </div>
  )
};
