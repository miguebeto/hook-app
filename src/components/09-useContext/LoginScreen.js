import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  //extraemos el estado y la funcion seteadora que provee el componente main mediante el context
  const {user, setUser} = useContext(UserContext);

  return (
      <div>
          <h1>Login Screen </h1>
          <hr />
          <button 
            className='btn btn-primary'
            onClick={ ()=> setUser({
              ...user,
              id: 123,
              name: 'Miguel',
            })}
            >
              Login
          </button>
      </div>
  )
};
