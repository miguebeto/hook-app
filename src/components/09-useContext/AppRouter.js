import React from 'react';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from "react-router-dom";

//creamos la rutas que serÃ¡n accesibles desde el menu mediante los link
export const AppRouter = () => {
  return (
      <Router>
          <div>
              <NavBar />
              <div className='container'>
              <Switch>
                  <Route exact path="/" component={ HomeScreen } />

                  <Route exact path="/about" component={ AboutScreen } />
                  <Route exact path="/login" component={ LoginScreen } />

                  {/* <Route component={ HomeScreen } /> */}
                  <Redirect to='/' />
              </Switch>
              </div>
          </div>
      </Router>
  )
};
