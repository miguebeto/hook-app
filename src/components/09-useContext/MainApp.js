import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
  //creamos el estado con initialState de un objeto vacio {} el cual ser√° pasado por el value a los consumidores que haran los cambios
  const [user, setUser] = useState({}); 

  return (
      <UserContext.Provider value={ { 
        user,
        setUser,
       } }>
        <AppRouter />
      </UserContext.Provider>
  )
};
