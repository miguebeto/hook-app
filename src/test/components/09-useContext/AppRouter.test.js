import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en el componente <AppRoute />', ()=> {

     //creamos el usario con la informacion que se va a proporcionar en el context para pruebas
     const user = {
        id: 1,
        name: 'Miguel'
    };


    //debe retornar el componente sin cambios en el render
    const wrapper = mount(
        <UserContext.Provider value={user}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('Debe mantenerse el componente sin cambios en el renderizado', ()=>{

        //comprueba que el componente se muestra correctamente
        expect(wrapper).toMatchSnapshot();
    });
});