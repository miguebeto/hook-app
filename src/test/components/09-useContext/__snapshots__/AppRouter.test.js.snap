// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AppRoute /> Debe mantenerse el componente sin cambios en el renderizado 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current={null}
        className="navbar-brand"
        href="/home"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre />
    </div>
  </div>
</div>
`;
